From a9668df44bd635d40b6e7b4db2d12e5cf91c8013 Mon Sep 17 00:00:00 2001
From: Steve Grubb <sgrubb@redhat.com>
Date: Thu, 5 Aug 2021 09:54:44 -0400
Subject: [PATCH] Add missing call to free_interpretation_list

---
 auparse/auparse.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/auparse/auparse.c b/auparse/auparse.c
index ee3c97b..18f1127 100644
--- a/auparse/auparse.c
+++ b/auparse/auparse.c
@@ -1,5 +1,5 @@
 /* auparse.c --
- * Copyright 2006-08,2012-19 Red Hat Inc., Durham, North Carolina.
+ * Copyright 2006-08,2012-19,21 Red Hat Inc.
  * All Rights Reserved.
  *
  * This library is free software; you can redistribute it and/or
@@ -2014,6 +2014,7 @@ const char *auparse_find_field_next(auparse_state_t *au)
 			r = aup_list_next(au->le);
 			if (r) {
 				aup_list_first_field(au->le);
+				free_interpretation_list();
 				load_interpretation_list(r->interp);
 			}
 		}
-- 

