From e95722695a1d3dc6ac266af8871069da74471999 Mon Sep 17 00:00:00 2001
From: Steve Grubb <sgrubb@redhat.com>
Date: Thu, 6 Oct 2022 13:34:15 -0400
Subject: Fix another krb5 memory leak

---
 src/auditd-listen.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/auditd-listen.c b/src/auditd-listen.c
index 61a3480..171974c 100644
--- a/src/auditd-listen.c
+++ b/src/auditd-listen.c
@@ -358,6 +358,7 @@ static int server_acquire_creds(const char *service_name,
 	if (major_status != GSS_S_COMPLETE) {
 		gss_failure("acquiring credentials",
 				major_status, minor_status);
+		(void) gss_release_name(&minor_status, &server_name);
 		return -1;
 	}
 
-- 
2.27.0

