From fc97c70fdba18280985747198a6ce836d39cce9e Mon Sep 17 00:00:00 2001
From: Steve Grubb <sgrubb@redhat.com>
Date: Sat, 7 Aug 2021 10:29:07 -0400
Subject: [PATCH 2196/2246] error out if log is mangled

---
 src/ausearch-parse.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ausearch-parse.c b/src/ausearch-parse.c
index b0c8b2a..81ef319 100644
--- a/src/ausearch-parse.c
+++ b/src/ausearch-parse.c
@@ -1995,6 +1995,10 @@ other_avc:
 			*term = '"';
 		} else { 
 			s->comm = unescape(str);
+			if (s->comm == NULL) {
+				rc = 11;
+				goto err;
+			}
 			term = str + 6;
 		}
 	}
-- 
1.8.3.1

