From ce58837d44b7d9fcb4e140c23f68e0c94d95ab6e Mon Sep 17 00:00:00 2001
From: Steve Grubb <sgrubb@redhat.com>
Date: Sat, 21 Aug 2021 10:20:11 -0400
Subject: [PATCH 2214/2246] When interpreting, if val is NULL return an empty
 string

---
 auparse/interpret.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/auparse/interpret.c b/auparse/interpret.c
index 177ab82..63829aa 100644
--- a/auparse/interpret.c
+++ b/auparse/interpret.c
@@ -840,6 +840,9 @@ static char *print_escaped(const char *val)
 {
 	char *out;
 
+	if (val == NULL)
+                        return strdup(" ");
+
         if (*val == '"') {
                 char *term;
                 val++;
-- 
1.8.3.1

