From d89e5647d9e090f45146c144d920bd1f686a8230 Mon Sep 17 00:00:00 2001
From: Steve Grubb <sgrubb@redhat.com>
Date: Thu, 15 Jul 2021 11:36:17 -0400
Subject: [PATCH 2163/2246] Move the free_config to success path

---
 src/auditd.c | 4 +++---
 1 file changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/auditd.c b/src/auditd.c
index ca69d3b..5478cc4 100644
--- a/src/auditd.c
+++ b/src/auditd.c
@@ -457,8 +457,10 @@ static int become_daemon(void)
 				return -1;
 
 			/* Success - die a happy death */
-			if (status == SUCCESS)
+			if (status == SUCCESS) {
+				free_config(&config);
 				_exit(0);
+			}
 			return -1;
 	}
 
-- 
1.8.3.1

